src/controller/Controller.cpp

```diff
@@ -32,46 +105,23 @@ void Controller::Read(OSContPad* pad, int32_t virtualSlot) {
     padToBuffer.button |= getPressedButtons(virtualSlot) & 0xFFFF;
 
     // Stick Inputs
-    if (getLeftStickX(virtualSlot) == 0) {
-        if (getPressedButtons(virtualSlot) & BTN_STICKLEFT) {
-            padToBuffer.stick_x = -128;
-        } else if (getPressedButtons(virtualSlot) & BTN_STICKRIGHT) {
-            padToBuffer.stick_x = 127;
-        }
-    } else {
-        padToBuffer.stick_x = getLeftStickX(virtualSlot);
-    }
+    int8_t leftStickX = ReadStick(virtualSlot, LEFT, X);
+    int8_t leftStickY = ReadStick(virtualSlot, LEFT, Y);
+    int8_t rightStickX = ReadStick(virtualSlot, RIGHT, X);
+    int8_t rightStickY = ReadStick(virtualSlot, RIGHT, Y);
 
-    if (getLeftStickY(virtualSlot) == 0) {
-        if (getPressedButtons(virtualSlot) & BTN_STICKDOWN) {
-            padToBuffer.stick_y = -128;
-        } else if (getPressedButtons(virtualSlot) & BTN_STICKUP) {
-            padToBuffer.stick_y = 127;
-        }
-    } else {
-        padToBuffer.stick_y = getLeftStickY(virtualSlot);
-    }
+    auto profile = getProfile(virtualSlot);
+    ProcessStick(leftStickX, leftStickY, profile->AxisDeadzones[0]);
```

Comment:
I believe the 0 and 2 here are SDL dependent values. The axis used to represent the x direction for the left and right stick aren't guaranteed to be the same for each controller backend (wii u, switch, etc)

Reply:
If we're removing the code that checks that the axis deadzones are separate, then we should also take in both the x and y deadzones for this function.

Reply:
I'm OK with the answer being that deadzones are not implemented separately yet, but we should keep the warnings for now since the deadzone for x and y should always be the same right now to create a circle.

You checked the backends that _we_ implement. Have you done any research on other controller backends like xinput?

Reply:
What I'm really asking is this: What if we have a backend like xinput that doesn't have the same range?

Reply:
Let's pass in both the x and y deadzone here even if we're going to ignore the y deadzone.